src/solvers/floatbv/float_utils.cpp

```diff
@@ -84,58 +83,57 @@ bvt float_utilst::to_integer(
-
-  const unbiased_floatt unpacked=unpack(src);
+  PRECONDITION(src.size() == spec.width());
 
   // The following is the usual case in ANSI-C, and we optimize for that.
-  if(rounding_mode_bits.round_to_zero.is_true())
-  {
-    bvt fraction=unpacked.fraction;
+  PRECONDITION(rounding_mode_bits.round_to_zero.is_true());
 
-    if(dest_width>fraction.size())
-    {
-      bvt lsb_extension=bv_utils.build_constant(0U, dest_width-fraction.size());
-      fraction.insert(fraction.begin(),
-                      lsb_extension.begin(),
-                      lsb_extension.end());
-    }
+  const unbiased_floatt unpacked = unpack(src);
 
-    // if the exponent is positive, shift right
-    bvt offset=bv_utils.build_constant(fraction.size()-1,
-                                       unpacked.exponent.size());
-    bvt distance=bv_utils.sub(offset, unpacked.exponent);
-    bvt shift_result=bv_utils.shift(
-      fraction, bv_utilst::shiftt::SHIFT_LRIGHT, distance);
+  bvt fraction = unpacked.fraction;
 
-    // if the exponent is negative, we have zero anyways
-    bvt result=shift_result;
-    literalt exponent_sign=unpacked.exponent[unpacked.exponent.size()-1];
+  if(dest_width > fraction.size())
+  {
+    bvt lsb_extension =
+      bv_utils.build_constant(0U, dest_width - fraction.size());
+    fraction.insert(
+      fraction.begin(), lsb_extension.begin(), lsb_extension.end());
+  }
 
-    for(std::size_t i=0; i<result.size(); i++)
-      result[i]=prop.land(result[i], !exponent_sign);
+  // if the exponent is positive, shift right
+  bvt offset =
+    bv_utils.build_constant(fraction.size() - 1, unpacked.exponent.size());
+  bvt distance = bv_utils.sub(offset, unpacked.exponent);
+  bvt shift_result =
+    bv_utils.shift(fraction, bv_utilst::shiftt::SHIFT_LRIGHT, distance);
 
-    // chop out the right number of bits from the result
-    if(result.size()>dest_width)
-    {
-      result.resize(dest_width);
-    }
-
-    assert(result.size()==dest_width);
+  // if the exponent is negative, we have zero anyways
+  bvt result = shift_result;
+  literalt exponent_sign = unpacked.exponent[unpacked.exponent.size() - 1];
 
-    // if signed, apply sign.
-    if(is_signed)
-      result=bv_utils.cond_negate(result, unpacked.sign);
-    else
-    {
-      // It's unclear what the behaviour for negative floats
-      // to integer shall be.
-    }
+  for(std::size_t i = 0; i < result.size(); i++)
+    result[i] = prop.land(result[i], !exponent_sign);
 
-    return result;
+  // chop out the right number of bits from the result
+  if(result.size() > dest_width)
+  {
+    result.resize(dest_width);
   }
+
+  INVARIANT(
+    result.size() == dest_width,
+    "result bitvector width should equal the destination bitvector width");
+
+  // if signed, apply sign.
+  if(is_signed)
+    result = bv_utils.cond_negate(result, unpacked.sign);
   else
-    throw "unsupported rounding mode";
+  {
+    // It's unclear what the behaviour for negative floats
```

Comment:
`UNIMPLEMENTED;` ?

Reply:
I think that's deliberate. According to the C standard, converting a negative float to an unsigned int is undefined behaviour. The code here chooses to simply ignore the sign bit when converting to an unsigned int (i.e., positive floats and their negative counterparts yield the same unsigned int value).

Reply:
It seems this case is covered by `--conversion-check`. 
Wouldn't it be better for a verifier to make the result of the conversion nondeterministic? (That question goes probably beyond this PR, though).

Reply:
I just tried this and `--conversion-check` indeed finds this error. Yes, to be more sound for when `--conversion-check` is not enabled we could make the conversion result nondet but I'd say we leave this as future work for now.